src/requests/completions.jl

```diff
@@ -63,106 +65,50 @@ function latex_completions(doc, offset, partial, CIs)
-        end
-        if startswith("local", spartial)
-            push!(CIs, CompletionItem("local", 14, "local", TextEdit(Range(doc, offset:offset), "local \$0\n"[length(spartial) + 1:end])))
-        end
-    elseif fc == 'm'
-        if startswith("macro", spartial)
-            push!(CIs, CompletionItem("macro", 14, "macro", TextEdit(Range(doc, offset:offset), "macro \$1(\$2)\n    \$0\nend"[length(spartial) + 1:end])))
-        end
-        if startswith("module", spartial)
-            push!(CIs, CompletionItem("module", 14, "module", TextEdit(Range(doc, offset:offset), "module \$0\nend"[length(spartial) + 1:end])))
-        end
-        if startswith("mutable", spartial)
-            push!(CIs, CompletionItem("mutable", 14, "mutable", TextEdit(Range(doc, offset:offset), "mutable struct \$1\n    \$0\nend"[length(spartial) + 1:end])))
-        end
-    elseif startswith("outer", spartial)
-        push!(CIs, CompletionItem("outer", 14, "outer", TextEdit(Range(doc, offset:offset), "outer"[length(spartial) + 1:end])))
-    elseif startswith("primitive", spartial)
-        push!(CIs, CompletionItem("primitive", 14, "primitive", TextEdit(Range(doc, offset:offset), "primitive type \$1\n    \$0\nend"[length(spartial) + 1:end])))
-    elseif startswith("quote", spartial)
-        push!(CIs, CompletionItem("quote", 14, "quote", TextEdit(Range(doc, offset:offset), "quote\n    \$0\nend"[length(spartial) + 1:end])))
-    elseif startswith("return", spartial)
-        push!(CIs, CompletionItem("return", 14, "return", TextEdit(Range(doc, offset:offset), "return \$0"[length(spartial) + 1:end])))
-    elseif startswith("struct", spartial)
-        push!(CIs, CompletionItem("struct", 14, "struct", TextEdit(Range(doc, offset:offset), "struct \$1\n    \$0\nend"[length(spartial) + 1:end])))
-    elseif fc == 't'
-        if startswith("try", spartial)
-            push!(CIs, CompletionItem("try", 14, "try", TextEdit(Range(doc, offset:offset), "try \$1\n    \$0\ncatch\nend"[length(spartial) + 1:end])))
-        end
-    elseif startswith("using", spartial)
-        push!(CIs, CompletionItem("using", 14, "using", TextEdit(Range(doc, offset:offset), "using \$0\n"[length(spartial) + 1:end])))
-    elseif startswith("while", spartial)
-        push!(CIs, CompletionItem("while", 14, "while", TextEdit(Range(doc, offset:offset), "while \$1\n    \$0\nend"[length(spartial) + 1:end])))
     end
 end
 
+const snippet_completions = Dict{String,String}(
+    "abstract" => "abstract type \$0 end",
+    "baremodule" => "baremodule \$0\nend",
+    "begin" => "begin\$0\nend", 
+    "break" => "break",
+    "catch" => "catch",
+    "const" => "const ",
+    "continue" => "continue",
+    "do" => "do \$0\nend",
+    "else" => "else",
+    "elseif" => "elseif ",
+    "end" => "end",
+    "export" => "export ",
+    "finally" => "finally",
+    "for" => "for \$0 in \$1\n    \$2\nend",
```

Comment:
Slightly fancier:

```suggestion
    "for" => "for \$3\${2|=, in |}\$1\n    \$0\nend",
```

Reply:
I'm going to adjust the PR to provide the most barebones implementation I can think of. 

Personally I would prefer a single placeholder for each entry as completions (etc) don't seem to be enabled when jumping between the tab stops. Thoughts on that?

Reply:
I also don't think `itr = Something` reads particularly well but that could just be me!

Reply:
I think completions not working when in the midst of filling a snippet might be just a vscode issue. That said, these kind of snippets aren't super useful compared to e.g. filling out function argument placeholders in the future if we generate snippets for specific functions/methods. I'm fine with leaving them *relatively* simple.